src/overlay/Peer.cpp

```diff
@@ -1237,6 +1584,22 @@ Peer::recvAuth(StellarMessage const& msg)
         return;
     }
 
+    // Subtle: after successful auth, must send sendMore message first to tell
+    // the other peer if this node prefers to flow control the traffic or not
+    if (mRemoteOverlayVersion >= Peer::FIRST_VERSION_SUPPORTING_FLOW_CONTROL)
+    {
+        if (mApp.getConfig().ENABLE_OVERLAY_FLOW_CONTROL)
+        {
+            sendSendMore(mApp.getConfig().PEER_FLOOD_READING_CAPACITY);
+        }
+        else
+        {
+            // 0 indicates peer doesn't want to use flow control
+            sendSendMore(0);
+        }
+    }
```

Comment:
I think you have a bug (missing test?):
you need to disable flow control in the else here as to avoid getting stuck in `DONT_KNOW` state